configure.ac

```diff
@@ -98,23 +97,45 @@ fi
 AM_CONDITIONAL(SYSTEMD, test -n "${with_systemdunitdir}")
 
 AC_ARG_WITH([dbuspolicydir],
-        AS_HELP_STRING([--with-dbuspolicydir=DIR], [D-Bus policy directory]),
-        [],
-        [with_dbuspolicydir="$($PKG_CONFIG --variable=datadir dbus-1)/dbus-1/system.d"])
+	    AS_HELP_STRING([--with-dbuspolicydir=DIR], [D-Bus policy directory]),
+	    [autodetect_dbuspolicydir=false],
+	    [autodetect_dbuspolicydir=true])
+
+AC_ARG_WITH([dbussystemservicedir],
+	    AS_HELP_STRING([--with-dbussystemservicedir=DIR], [D-Bus system service directory]),
+	    [autodetect_dbussystemservicedir=false],
+	    [autodetect_dbussystemservicedir=true])
+
+AC_ARG_WITH([dbusinterfacesdir],
+	    AS_HELP_STRING([--with-dbusinterfacesdir=DIR], [D-Bus interfaces directory]),
+	    [autodetect_dbusinterfacesdir=false],
+	    [autodetect_dbusinterfacesdir=true])
+
+AS_IF([$autodetect_dbuspolicydir || $autodetect_dbussystemservicedir || $autodetect_dbusinterfacesdir], [
+       PKG_CHECK_MODULES([DBUS], [dbus-1])
```

Comment:
Previously, this was only checked when the service is enabled. Should the whole block be conditional?

Reply:
right, before my patch `PKG_CHECK_MODULES(..., [dbus-1])` was only called when `--enable-service` was passed to `./configure`. However the usage of all the variables are not conditionally in `Makefile.am`, so this was buggy before.